### ISSUE TITLE
add tune-lock tile

### ISSUE THREAD

**Issue by corneum** [ddnet/ddnet#4651]
a new tune tile that doesnt work as a zone but as one of the power up tiles that give a tee a property after they have moved through it,
as i understand it cant be too hard to add because all of the hard stuff is already made. (tunes already exist & tuned tees are predicted now)
as a mapper i can honestly tell that this would be a massive gamechanger. there are countless possibilities.

**Comment by def-** [comment]
@trml would this predict properly?

**Comment by fokkonaut** [comment]
Hard to predict, because each tee has it's own set of tunings then. Too much for the snapshots definitely, and as netmessages it would be too overkill too, at least if the client knows the tunes of every player in their view range.
This needs some thinking probably.

**Comment by trml** [comment]
Without network changes it wouldn't (for other players), since each player only gets sent one tuning now, and you also don't get tunings from other tunzeones without being in them.

If lock-tunes should override other tunezones one would also need some kind of flag for each possible zone (or tune setting) to tell if that tuning is locked for the player.

One option would perhaps be to send a bitarray with which zones a player is locked to (by associating a tune-lock tile with a tunezone, and use it to lock all tunes in that zone), although the most flexible for modding would perhaps be to send the tuning of all players.

**Comment by fokkonaut** [comment]
Yes. Also, I think what could happen is tile 1 sets 3 tunes, and tile 2 changes one of those 3 again. So we should implement this carefully.

**Comment by corneum** [comment]
but right now you can also tune the server settings and still have tunezones. they just override the server. 
no mixing even if 2 tunes appear multiple times in different zones. if you enter a tunezone it should tune the standart server settings not the settings that the tee already has.

it would also be cool if there was a easy tile for setting everything back to normal. maybe any zone without tunes?

**Comment by C0D3D3V** [comment]
See https://github.com/ddnet/ddnet/pull/4691 for a first implementation
